### PR TITLE
🌱Remove go-bindata

### PR DESCRIPTION
**Pull request by fabriziopandini**
**What this PR does / why we need it**:
This PR is a follow up of https://github.com/kubernetes-sigs/cluster-api/pull/4350 and it removes the remaining go-bindata bits in favour of native golang "embed" introduced in v1.16.

**Which issue(s) this PR fixes**

### CODE REVIEW

File: test/framework/kubernetesversions/zz_generated.bindata.go [hunk 1]
```diff
@@ -1,285 +0,0 @@
-/*
```
Comment (CecileRobertMichon) [inline test/framework/kubernetesversions/zz_generated.bindata.go:1]: was this removed intentionally? I don't see it being replaced with embed in this PR

Comment (fabriziopandini) [reply test/framework/kubernetesversions/zz_generated.bindata.go:1]: This was replaced by go embed in the previous PR, those are left-overs

**Comment by fabriziopandini** [comment]
:sad I have to investigate why Kustomise on my machine splits descriptions on many line, while in CI not...

**Comment by vincepri** [comment]
@fabriziopandini Might want to try to `make clean-bins`?

**Comment by fabriziopandini** [comment]
@vincepri this should be ready to go

**Review by vincepri — APPROVED** [review]
/approve
/lgtm

Thank you!

**Pull request merged**